packages/block-editor/src/hooks/test/font-size.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * External dependencies
+ */
+import { renderHook } from '@testing-library/react';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter, removeFilter } from '@wordpress/hooks';
+import {
+	getBlockTypes,
+	registerBlockType,
+	unregisterBlockType,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import _fontSize from '../font-size';
+
+const noop = () => {};
+
+describe( 'useBlockProps', () => {
+	const blockSettings = {
+		save: () => noop,
+		category: 'text',
+		title: 'font size title',
+		name: 'test/font-size',
+		supports: {
+			typography: {
+				fontSize: true,
+			},
+		},
+	};
+
+	afterEach( () => {
+		getBlockTypes().forEach( ( block ) => {
+			unregisterBlockType( block.name );
+		} );
+	} );
+
+	it( 'should return preset classname', () => {
+		registerBlockType( blockSettings.name, blockSettings );
+		addFilter(
+			'blockEditor.useSetting.before',
+			'test/useSetting.before',
+			( result, path ) => {
+				if ( 'typography.fontSizes' === path ) {
+					return [
+						{
+							name: 'A larger font',
+							size: '32px',
+							slug: 'larger',
+						},
+					];
+				}
+
+				if ( 'typography.fluid' === path ) {
+					return false;
+				}
+
+				if ( 'layout' === path ) {
+					return {};
+				}
+
+				return result;
+			}
+		);
+
+		const { result } = renderHook( () =>
+			_fontSize.useBlockProps( {
+				name: blockSettings.name,
+				fontSize: 'larger',
+			} )
+		);
+		const { style, className } = result.current;
+		expect( className ).toBe( 'has-larger-font-size' );
+		expect( style.fontSize ).toBe( '32px' );
+
+		removeFilter(
+			'blockEditor.useSetting.before',
+			'test/useSetting.before'
+		);
```

Comment:
Tiniest of nits: could each of these `removeFilter` calls go in `afterEach`? Just thinking that way they'd fire even if the `expect` fails, and since it looks like the calls don't need to reference the callback.

Not a blocker, though!

Reply:
Thanks for testing! You got it! I'll fix this before merge. If not later, then tomorrow.